src/cache/server/cache.go

```diff
@@ -140,6 +140,23 @@ func (cache *Cache) removeAndDeleteFile(p string, file *cachedFile) {
 // return whatever's been stored there, which might be a directory and therefore contain
 // multiple files to be returned.
 func (cache *Cache) RetrieveArtifact(artPath string) (map[string][]byte, error) {
+	ret := map[string][]byte{}
+	if strings.Contains(artPath, "*") {
```

Comment:
Curious, is this how we detect we are globbing? Globs don't always need to contain that wildcard right? (It could equally be something like "?" etc right?